app/src/main/java/com/example/task/adapters/CompletedAdapter.kt

```diff
@@ -35,9 +32,24 @@ class CompletedAdapter(private val task: List<Task>, private val onClick: (Task,
         holder.radioButton.setOnClickListener{
             onClick(task[holder.adapterPosition],ClickType.SHORT,it)
         }
-
         holder.taskList.setOnLongClickListener {
-            onClick(task[holder.adapterPosition], ClickType.LONG,it)
+            val popupMenu: PopupMenu = PopupMenu(it.context,it)
+            popupMenu.menuInflater.inflate(R.menu.item_list_menu,popupMenu.menu)
+            popupMenu.setOnMenuItemClickListener {
+                when(it.itemId){
+                    R.id.delete -> {
+                        onClick(task[holder.adapterPosition], ClickType.LONG_DELETE, holder.itemView)
+                        true
+                    }
+                    R.id.edit -> {
+                        Toast.makeText(holder.itemView.context,"Can't edit the task after completion",Toast.LENGTH_LONG).show()
+                        true
+                    }
+                    else -> true
+                }
+            }
+            popupMenu.show()
+
```

Comment:
Hide edit option here. No need to show it. Use findViewById and set Visibility to Gone.